packages/js/src/utils/typescript/ts-solution-setup.ts

```diff
@@ -290,14 +291,15 @@ export function getProjectType(
 }
 
 export function getProjectSourceRoot(
-  tree: Tree,
-  projectSourceRoot: string | undefined,
-  projectRoot: string
-): string | undefined {
+  project: ProjectConfiguration,
+  tree?: Tree
+): string {
+  tree ??= new FsTree(workspaceRoot, false);
```

Comment:
Not really a fan of creating a FsTree just for this. Is this `tree` even needed? I think we can use 2 different `exists` functions here rather than creating a tree?